File: server/Gacha/banner.cs
namespace Nanina.Gacha
{
    public class BannerPoolElement
    {
        public double weight;
        public required string waifuId;
    }
    public class Banner {
        public required string id;
        public short pityAmount;
        public short pullCost;
        public required List<BannerPoolElement> standardPool;
        public required List<BannerPoolElement> pityPool;
    }
}

File: client/src/classes/banner/banner.ts
import type BannerPoolElement from "./banner_pool_element";

export default class Banner {
    public id : string = "";
    public pityAmount : number = 0;
    public pullCost : number = 0;
    public standardPool !: BannerPoolElement[]
    public  pityPool !: BannerPoolElement[]
}